File: udemy/pascal/projetos/SE_triangulo.pas
program triangulo;
var
    l1, l2, l3: real; 
begin
  write('informe o lado 1: ');
  read(l1);
  write('informe o lado 2: ');
  read(l2);
  write('informe o lado 3: ');
  read(l3);
    if ((l1+l2)>l3) and ((l2+l3)> l1) and ((l3+l1)>l2) then
        if (l1 = l2) and (l2= l3) then
            write('os lados formam um triangulo equilatero')
        else if (l1 = l2) or (l2 = l3) or (l3 = l1) then 
            write('os lados formam um triangulo isosceles')
        else 
            write('os lados formam um triangulo escaleno')
    else 
        write('os lados nao formam um triangulo');
end.

File: udemy/java/SE_triangulo/src/Se_triangulo/SE_triangulo.java
package Se_triangulo;

import javax.swing.JOptionPane;

public class SE_triangulo {

	public static void main(String[] args) {
		float l1,l2,l3;
		String mensagem= "";
		l1 = Float.parseFloat(JOptionPane.showInputDialog("informe o primeiro lado do triangulo:  "));
		l2 = Float.parseFloat(JOptionPane.showInputDialog("informe o segundo lado do triangulo:  "));
		l3 = Float.parseFloat(JOptionPane.showInputDialog("informe o terceiro lado do triangulo:  "));
		if (((l1+l2)>l3) && ((l2+l3)>l1) && ((l3+l1)>l2)) {
			mensagem = mensagem + "os lados formam um triangulo ";
			if ((l2 == l1) && (l2 == l3)) {
				mensagem = mensagem + "equilatero";
			}
			else if ((l1 == l2) || (l2==l3) ||(l3==l1)){
				mensagem = mensagem + "isosceles";
			}
			else {
				mensagem = mensagem + "escaleno";
			}	
		}
		else {
			mensagem = mensagem + "os lados não formam um triangulo";
		}
		JOptionPane.showMessageDialog(null, mensagem);
	}

}
